dms-domain/src/main/kotlin/team/aliens/dms/domain/remain/model/RemainAvailableTime.kt

```diff
@@ -16,6 +17,27 @@ data class RemainAvailableTime(
 
     val endDayOfWeek: DayOfWeek,
 
-    val endTime: LocalTime,
+    val endTime: LocalTime
 
-)
\ No newline at end of file
+) {
+
+    fun isAvailable(): Boolean {
+        val currentDateTime = LocalDateTime.now()
+        val dayOfWeek = currentDateTime.dayOfWeek.value
+        val now = currentDateTime.toLocalTime()
+
+        if (dayOfWeek < startDayOfWeek.value || dayOfWeek > endDayOfWeek.value) {
+            return false
+        }
+
+        if (dayOfWeek == startDayOfWeek.value && now < startTime) {
+            return false
+        }
+
+        if (dayOfWeek == endDayOfWeek.value && endTime < now) {
+            return false
+        }
```

Comment:
p2; 하나의 함수에 return 문이 너무 많은 것 같습니다.
조건문을 별도의 함수로 분리해 용도를 쉽게 파악하고 코드를 간결하게 만들면 좋을 것 같습니다.

ex)
```suggestion
        if (isOutOfRangeDay() || isBeforeStartTime() || isAfterEndTime()) {
            return false
        }
```